### PR TITLE
Lesson6

### PR DESCRIPTION
**Pull request by shrutiyer**

### CODE REVIEW

**Comment by davidabrahams** [comment]
Yayyyyyy. Will review eventually. <3

File: lesson6/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+#### 2. Design
+Asking question while looking at an app can be incredibly helpful while trying to design something. Once you have an idea what your app is going to be about, make a first rought draft. Concentrate on the features you will be providing and the top-level navigation of the app. 
+
+Draw the screens on pieces of paper (or use one of the many online programs like fluidui). Test these mockups on your friends. It is recommended that you let your friends first explore the mockups without any instructions and observe what they see, think and do. Later, you can ask more specific questions about the features in your mockup.
+
+Next step is to integrate whatever feedback you received from the User Testing into your app design. At this stage, you can also decide on the color scheme of the app, logo, name etc. You could do cycles of user testing-->feedback integration--> testing again until you are satisfied with the app.
+
+Apps like FB Messenger or Snapchat would go through many periodic Usability Testing. It is important to make sure that your app remains sleek and intuitive while adding new features or editing existing features.
```
Comment (davidabrahams) [inline lesson6/README.md]: grammar

File: lesson6/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+#### 3. Testing after developing:
+Once you have a Minimum Viable Product (MVP) of your app, test again. But this time, it is a little different.
+  1. Try to break it
+      * Rotate the device. Do all elements adjust well?
+      * Tap at random places
+      * Try devices of different screen sizes.
```
Comment (davidabrahams) [inline lesson6/README.md:31]: Try different hardware. Try different OSs.

File: lesson6/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  2. Get a fresh pair of eyes
+  3. Test your code for edge cases and weird events 
+
+## App Elements
+Note: The documentation for all the app elements listed below are really good. We tried to summarize it so that you get an idea of each of the elements. When you use them in your app, the developers website is the best place to learn more about them.
+
+#### Layouts
+While writing the XML files for your app, you might have noticed there are numerous ways to layout and display the content. [This](https://developer.android.com/guide/topics/ui/declaring-layout.html) document explains layouts really well and talks about margins, padding, . Go read it!
```
Comment (davidabrahams) [inline lesson6/README.md]: Maybe make this reading part of HW

File: lesson6/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  * Access to important actions in a predictable way, such as search.
+  * Support for navigation and view switching (with tabs or drop-down lists).
+
+#### [Settings](https://developer.android.com/guide/topics/ui/settings.html)
+Remember how one of the design principles said "Let me make it mine"? We want to provide them with good defaults but always provide easy ways to change those defaults. Settings page just does that.
+
+If you want to learn about how to design a Settings page, [read this guide](https://material.google.com/patterns/settings.html#settings-usage).
+
+#### [Pop-up messages](https://developer.android.com/training/snackbar/index.html)
+Whenever you need to provide the users with a prominent enough message, use pop-up messages. Most of the time, the user doesn't have to respond to the message. Also, the message disappears after a short while. The pop-up message should not block the users from using the app.
```
Comment (davidabrahams) [inline lesson6/README.md:63]: Maybe detail how to do this? (What does a non-intrusive non-blocking pop up look like?)

File: lesson6/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Open any app on your phone. It could be very simple or very complex; there is something to learn from both. While using it, ask questions. Some examples of questions could be:
+  1. What does the logo/name communicate to the users?
+  2. Are all the features easily discoverable?
+  3. In order to perform an action, how many times did you tap? Too many popups asking you to choose YES or NO?
+  4. How does the color scheme fit the overall feel?
+
+Read through the [Android Design Principles](https://developer.android.com/design/get-started/principles.html). It will help you think of more questions to ask.
+
+#### 2. Design
+Asking question while looking at an app can be incredibly helpful while trying to design something. Once you have an idea what your app is going to be about, make a first rought draft. Concentrate on the features you will be providing and the top-level navigation of the app.
```
Comment (davidabrahams) [inline lesson6/README.md]: questions

File: lesson6/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  2. [User Interface Guide](https://developer.android.com/guide/topics/ui/index.html)
+  3. [Material Design](https://material.google.com/)
+  4. [Android Design Principles](https://developer.android.com/design/get-started/principles.html)
+  5. [Material Design apps, icons, etc](https://material.uplabs.com/)
+
+### Try It Out!
+ 
+**[HW]** 
+  1. Download Facebook app on an Android phone (if you don't have access to either a Facebook account or an Android phone, let the teaching team know)
+  2. Observe and ask questions about the design intention. The design principles will help you here. Write up the questions and the corresponding findings?
```
Comment (davidabrahams) [inline lesson6/README.md]: What design principles?

File: lesson6/README.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  3. You might have thoughts about a different layout. Prototype on paper. Include your prototype's picture (and its revisions) in your write-up. You could also use one of the programs (like Fluid UI) to make the prototypes.
```
Comment (davidabrahams) [inline lesson6/README.md:88]: A different layout for the FB app?

File: lesson6/README.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  1. [Best Practices for User Interface](https://developer.android.com/training/best-ui.html)
+  2. [User Interface Guide](https://developer.android.com/guide/topics/ui/index.html)
+  3. [Material Design](https://material.google.com/)
+  4. [Android Design Principles](https://developer.android.com/design/get-started/principles.html)
+  5. [Material Design apps, icons, etc](https://material.uplabs.com/)
+
+### Try It Out!
+ 
+**[HW]** 
+  1. Download Facebook app on an Android phone (if you don't have access to either a Facebook account or an Android phone, let the teaching team know)
```
Comment (davidabrahams) [inline lesson6/README.md]: Should we let them choose what app (FB, twitter, instagram, etc)? Maybe ask them to do 2-3 and compare them?

File: lesson6/README.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  2. Observe and ask questions about the design intention. The design principles will help you here. Write up the questions and the corresponding findings?
+  3. You might have thoughts about a different layout. Prototype on paper. Include your prototype's picture (and its revisions) in your write-up. You could also use one of the programs (like Fluid UI) to make the prototypes.
+  4. (Optional) Recreate your layout(s) in Android Studio by writing XML file(s) for them.
```
Comment (davidabrahams) [inline lesson6/README.md]: Like, make their previous apps pretty? I think this should be mandatory...

**Review by davidabrahams — CHANGES_REQUESTED** [review]
Maybe you should make a [HW] tag for the parts they have to do, so they know what "getting through the README" means.

File: lesson6/README.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  
+Fill out [this survey](https://goo.gl/forms/LLUhDfZzBkkqawtp1) after you finish the HW
```
Comment (davidabrahams) [inline lesson6/README.md:91]: Assuming they have to change their code, we should ask for a link to the repo where their code is (unless we just assume its in their class repo)

**Review by nmohamed — CHANGES_REQUESTED** [review]
do this

File: lesson6/README.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
+# Lesson 6 - How to Create an App
+
+##  Designing an app
+
```
Comment (nmohamed) [inline lesson6/README.md:4]: maybe preface this with a short intro to what this lesson is about? also specify the due date of it above

File: lesson6/README.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+#### 1. Observe
+Open any app on your phone. It could be very simple or very complex; there is something to learn from both. While using it, ask questions. Some examples of questions could be:
+  1. What does the logo/name communicate to the users?
+  2. Are all the features easily discoverable?
+  3. In order to perform an action, how many times did you tap? Too many popups asking you to choose YES or NO?
+  4. How does the color scheme fit the overall feel?
```
Comment (nmohamed) [inline lesson6/README.md:12]: how about "how easy are certain buttons to press? do you press those buttons often?"

File: lesson6/README.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+Read through the [Android Design Principles](https://developer.android.com/design/get-started/principles.html). It will help you think of more questions to ask.
+
+#### 2. Design
+Asking question while looking at an app can be incredibly helpful while trying to design something. Once you have an idea what your app is going to be about, make a first rought draft. Concentrate on the features you will be providing and the top-level navigation of the app. 
+
+Draw the screens on pieces of paper (or use one of the many online programs like fluidui). Test these mockups on your friends. It is recommended that you let your friends first explore the mockups without any instructions and observe what they see, think and do. Later, you can ask more specific questions about the features in your mockup.
```
Comment (nmohamed) [inline lesson6/README.md]: paint, photoshop, whiteboards, your approved medium of art

File: lesson6/README.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+Next step is to integrate whatever feedback you received from the User Testing into your app design. At this stage, you can also decide on the color scheme of the app, logo, name etc. You could do cycles of user testing-->feedback integration--> testing again until you are satisfied with the app.
```
Comment (nmohamed) [inline lesson6/README.md]: ➔

File: lesson6/README.md (hunk 1 above)
Comment (nmohamed) [inline lesson6/README.md]: spell out Facebook
why is Usability Testing capitalized? grammar for that sentence

File: lesson6/README.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+
+Apps like FB Messenger or Snapchat would go through many periodic Usability Testing. It is important to make sure that your app remains sleek and intuitive while adding new features or editing existing features.
+
+#### 3. Testing after developing:
+Once you have a Minimum Viable Product (MVP) of your app, test again. But this time, it is a little different.
```
Comment (nmohamed) [inline lesson6/README.md]: *test it again.
Don't really need the "But this time, it is a little different."
Maybe mention trying to test it after adding little things

File: lesson6/README.md (hunk 6 above)
Comment (nmohamed) [reply lesson6/README.md]: remove question mark

File: lesson6/README.md (hunk 11 above)
Comment (nmohamed) [reply lesson6/README.md:4]: also, it's a bit confusing reading through this. when i started it sounds like you're giving instructions to me, so it seemed like the entire thing is homework

**Review by nmohamed — CHANGES_REQUESTED** [review]
hi

**Review by nmohamed — APPROVED** [review]

File: lesson6/README.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
+# Lesson 6 - How to Design an App
+
+##  Designing an app
+
+Here is an example process for desigining an app:
+
+#### 1. Observe
+Open up any app on your phone. It could be very simple or very complex; there's something to learn from both. While using it, ask questions. Some examples of questions could be:
```
Comment (billmwong) [inline lesson6/README.md]: Should it only be an android app? IOS apps have slightly different design principles.  Or is this guide for general design, not just android design?

File: lesson6/README.md (hunk 9 above)
Comment (billmwong) [reply lesson6/README.md]: Yeah, I feel like there has to be some sort of coding/do it yourself in android studio portion of the assignment

**Review by billmwong — CHANGES_REQUESTED** [review]

File: lesson6/README.md (hunk 4 above)
Comment (shrutiyer) [reply lesson6/README.md:63]: There is a picture in the readme that shows the non-intrusive pop-up. Do we need to show anything more than that?

File: lesson6/README.md (hunk 8 above)
Comment (shrutiyer) [reply lesson6/README.md]: I wanted to choose a complicated app where too many things are happening. FB has way more stuff than twitter. I haven't used Instagram

File: lesson6/README.md (hunk 6 above)
Comment (shrutiyer) [reply lesson6/README.md]: Android design principles mentioned above

File: lesson6/README.md (hunk 7 above)
Comment (shrutiyer) [reply lesson6/README.md:88]: yep

File: lesson6/README.md (hunk 9 above)
Comment (shrutiyer) [reply lesson6/README.md]: Made it required

**Comment by shrutiyer** [comment]
Made more small changes. Comment on the new stuff.

**Review by billmwong — APPROVED** [review]
Approved!

**Review by davidabrahams — APPROVED** [review]
